apis/v1alpha1/defaulting.go

```diff
@@ -211,6 +276,50 @@ func (in *GreptimeDBCluster) mergeFlownodeTemplate() error {
 	return nil
 }
 
+func (in *GreptimeDBCluster) mergeLogging() error {
+	if logging := in.GetMeta().GetLogging(); logging != nil {
+		if err := mergo.Merge(logging, in.GetLogging().DeepCopy()); err != nil {
+			return err
+		}
+		if in.GetMonitoring().IsEnabled() {
+			// Set the default logging format to JSON if monitoring is enabled.
+			logging.Format = LogFormatJSON
+		}
+	}
+
+	if logging := in.GetDatanode().GetLogging(); logging != nil {
+		if err := mergo.Merge(logging, in.GetLogging().DeepCopy()); err != nil {
+			return err
+		}
+		if in.GetMonitoring().IsEnabled() {
+			// Set the default logging format to JSON if monitoring is enabled.
+			logging.Format = LogFormatJSON
```

Comment:
Users can set the log format to `text` or `json` ？

Reply:
If monitoring is enabled, it should be `JSON`, no matter what the user sets.